compliance_checker/cf/cf.py

```diff
@@ -1323,23 +1323,24 @@ def check_data_types(self, ds):
         total = len(ds.variables)
 
         for k, v in ds.variables.items():
-            if v.dtype.kind != "S" and all(
-                v.dtype.type != t
-                for t in (
-                    np.character,
-                    np.dtype("|S1"),
-                    np.dtype("b"),
-                    np.dtype("i2"),
-                    np.dtype("i4"),
-                    np.float32,
-                    np.double,
-                )
-            ):
-                fails.append(
-                    "The variable {} failed because the datatype is {}".format(
-                        k, v.datatype
+            if v.dtype is not str:
```

Comment:
I might be reading the indents wrong, but since this is an `if` statement followed by another `if` statement with no `else` conditions, I would recommend using a single `if` statement in lieu of nested `if`s to improve readability slightly.
```python
if v.dtype is not str \
        and v.dtype.kind != 'S' \
        and all(
            v.dtype.type != t
            for t in (
                np.character,
                np.dtype("|S1"),
                np.dtype("b"),
                np.dtype("i2"),
                np.dtype("i4"),
                np.float32,
                np.double
            )
        ):
    fails.append(
        "The variable {} failed because the datatype is {}".format(
            k,
            v.datatype,
        )
    )
```

Reply:
Thanks. Would be better to merge both if statements. I will update the PR.

Reply:
PR updated accordingly